pkg/statistics/handle/storage/read.go

```diff
@@ -43,7 +43,20 @@ import (
 
 // StatsMetaCountAndModifyCount reads count and modify_count for the given table from mysql.stats_meta.
 func StatsMetaCountAndModifyCount(sctx sessionctx.Context, tableID int64) (count, modifyCount int64, isNull bool, err error) {
-	rows, _, err := util.ExecRows(sctx, "select count, modify_count from mysql.stats_meta where table_id = %?", tableID)
+	return statsMetaCountAndModifyCount(sctx, tableID, false)
+}
+
+// StatsMetaCountAndModifyCountForUpdate reads count and modify_count for the given table from mysql.stats_meta with lock.
+func StatsMetaCountAndModifyCountForUpdate(sctx sessionctx.Context, tableID int64) (count, modifyCount int64, isNull bool, err error) {
+	return statsMetaCountAndModifyCount(sctx, tableID, false)
+}
+
+func statsMetaCountAndModifyCount(sctx sessionctx.Context, tableID int64, forUpdate bool) (count, modifyCount int64, isNull bool, err error) {
+	sql := "select count, modify_count from mysql.stats_meta where table_id = %?"
```

Comment:
The ` sctx.GetSessionVars().TxnMode` is empty for internal sessions(which means optimistic by default), I'm afraid the `for update` may be omitted here if we don't use `begin pessimistic`.

Reply:
<img width="984" alt="image" src="https://github.com/user-attachments/assets/ade12e61-df22-4e15-8511-f8b23e42797d">

We use ```WrapTxn``` to start running the transaction. so it will not lead to problem.